src/main/java/org/craftercms/studio/impl/v2/service/cluster/ClusterManagementServiceImpl.java

```diff
@@ -81,6 +93,50 @@ public boolean removeMembers(List<Long> memberIds) throws SiteNotFoundException
         return toRet;
     }
 
+    @Override
+    @HasPermission(type = DefaultPermission.class, action = "update_cluster")
+    public boolean setClusterPrimary() {
+        logger.debug("Enabling global publishing flag.");
+        studioClusterUtils.enableGlobalPublishing();
+        logger.debug("Enabled global publishing flag with result: " + studioClusterUtils.isGlobalPublishingEnabled());
+
+        if (studioClusterUtils.memberPrimaryPublisher()) {
+            logger.debug("Member has already been the primary publisher. No action occurs.");
+            return true;
+        }
+
+        // set member as primary publisher
+        String localAddress = studioClusterUtils.getClusterNodeLocalAddress();
+        ClusterMember localNode = clusterManagementServiceInternal.getMemberByLocalAddress(localAddress);
+        logger.debug("Set this node as primary publisher.");
+        retryingOperationFacade.setClusterPrimary(localNode.getId());
+
+        // add audit log
+        try {
+            SiteFeed siteFeed = siteService.getSite(studioConfiguration.getProperty(CONFIGURATION_GLOBAL_SYSTEM_SITE));
+            AuditLog auditLog = auditServiceInternal.createAuditLogEntry();
+            auditLog.setSiteId(siteFeed.getId());
+            auditLog.setOperation(OPERATION_SET_CLUSTER_PRIMARY);
+            auditLog.setActorId(securityService.getCurrentUser());
+            auditLog.setPrimaryTargetId(siteFeed.getSiteId());
+            auditLog.setPrimaryTargetType(TARGET_TYPE_CLUSTER_NODE);
+            auditLog.setPrimaryTargetValue(siteFeed.getName());
+            List<AuditLogParameter> parameters = new ArrayList<>();
+            AuditLogParameter parameter = new AuditLogParameter();
+            parameter.setTargetId(Long.toString(localNode.getId()));
+            parameter.setTargetType(TARGET_TYPE_CLUSTER_NODE);
+            parameter.setTargetValue(localNode.getLocalAddress());
+            parameters.add(parameter);
+
+            auditLog.setParameters(parameters);
+            auditServiceInternal.insertAuditLog(auditLog);
+        } catch (SiteNotFoundException e) {
+            logger.error("Error while adding audit log.", e);
```

Comment:
We really don't want to get to this exception.

Reply:
The site is `studio_root` so normally we don't have the exception.